### PR TITLE
[SYCL] Add a test for group sorting algorithm

### PR DESCRIPTION
**Pull request by andreyfe1**
Implementation of sorting algorithm is here: https://github.com/intel/llvm/pull/4439

Cases that were added:
Data types: 
*	uint32_t
*	double
*	sycl::half
*	uint16_t
*	size_t
*	int8_t
*	int32_t
*	float
*	custom type
Functors:
*	less
*	greater
*	custom
Sorters:
*	default
*	custom
Sorting function:
*	joint
*	over_group
Interfaces
*	sorter
*	compare
*	w/o compare or sorter
Data size:
*	1 group
*	multiple groups
Generic group:
*	work-group

Cases that were not added:
* sub-group

Signed-off-by: Fedorov, Andrey <andrey.fedorov@intel.com>

### CODE REVIEW

**Comment by andreyfe1** [comment]
Inviting @romanovvlad for review. Feel free to invite more people if needed

File: SYCL/GroupAlgorithm/SYCL2020/sort.cpp
Comment (v-klochkov) [inline SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: I believe spirv-max-version was already set to 1.3 long ago: https://github.com/intel/llvm/commit/0a67b10f2ca0d32be5d08938160422527c42e5f9

File: SYCL/GroupAlgorithm/SYCL2020/sort.cpp
Comment (Pennycook) [inline SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: Is this left over from a previous version?

File: SYCL/GroupAlgorithm/SYCL2020/sort.cpp
Comment (andreyfe1) [reply SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: ok. I'll remove it. Just "copy-paste" issue

File: SYCL/GroupAlgorithm/SYCL2020/sort.cpp
Comment (andreyfe1) [reply SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: Sorry, which version do you mean by previous?

Comment (Pennycook) [reply SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: Sorry for being confusing.  What I'm really asking is: Why is this comment here?

Comment (andreyfe1) [reply SYCL/GroupAlgorithm/SYCL2020/sort.cpp]: Ah, I see. The comment is to show that the `get_pointer` way also works. I can add a comment about it

**Comment by andreyfe1** [comment]
Hi, all. Feel free to approve and merge if you don't have any more comments to the patch

**Comment by romanovvlad** [comment]
> 
> 
> Hi, all. Feel free to approve and merge if you don't have any more comments to the patch

Currently the test fails in pre-commit - github UI doesn't allow to merge the patch. The implementation of the API this test validates should be merged first.

**Comment by andreyfe1** [comment]
@romanovvlad, implementation was merged to the `intel/llvm` repo. CI is passed except Windows testing. Seems, failures are not related to the changes in this patch. What are the next step for this PR? Waiting until fix for Windows is not merged separately?

**Comment by andreyfe1** [comment]
@Pennycook, do you have any other comments/questions related to the PR?

**Review by Pennycook — APPROVED** [review]

**Pull request merged**
